watchtower/wtclient/backup_task.go

```diff
@@ -151,13 +168,28 @@ func (t *backupTask) bindSession(session *wtdb.ClientSessionBody) error {
 		// underestimate the size by one byte. The diferrence in weight
 		// can cause different output values on the sweep transaction,
 		// so we mimic the original bug and create signatures using the
-		// original weight estimate.
-		weightEstimate.AddWitnessInput(
-			input.ToLocalPenaltyWitnessSize - 1,
-		)
+		// original weight estimate. For anchor channels we'll go ahead
+		// an use the correct penalty witness when signing our justice
+		// transactions.
+		if t.chanType.HasAnchors() {
+			weightEstimate.AddWitnessInput(
+				input.ToLocalPenaltyWitnessSize,
```

Comment:
Hm, yeah this is something I must keep in mind when changing weight estimate constants: #4775

Reply:
yeah, definitely a little subtle that some of these are now fixed params in the wt protocol. will double check that #4775 doesn't touch any of the required constants for anchors